eclipse-collections/src/main/java/org/eclipse/collections/impl/map/mutable/primitive/ObjectBooleanHashMapWithHashingStrategy.java

```diff
@@ -740,6 +740,20 @@ public boolean getIfAbsentPut(K key, boolean value)
         return value;
     }
 
+    @Override
+    public boolean getAndPut(K key, boolean putValue, boolean defaultValue)
+    {
+        int index = this.probe(key);
+        if (ObjectBooleanHashMapWithHashingStrategy.isNonSentinel(this.keys[index]) && this.nullSafeEquals(this.toNonSentinel(this.keys[index]), key))
+        {
+            boolean existingValue = this.values.get(index);
+            this.addKeyValueAtIndex(key, putValue, index);
```

Comment:
Can this just be `this.values.set(index, putValue);`?

Reply:
Fixed.